EngineLayer/ProteinParsimony/ProteinGroup.cs

```diff
@@ -379,6 +393,38 @@ public void CalculateSequenceCoverage()
                 }
             }
 
+            foreach (var protein in ListOfProteinsOrderedByAccession)
+            {
+                HashSet<int> fragmentCoveredResidues = new HashSet<int>();
+
+                foreach (var psm in proteinsWithUnambigSeqPsmsPSM[protein])
+                {
+                    if (psm.FragmentCoveragePositionInProtein != null)
+                    {
+                        foreach (int i in psm.FragmentCoveragePositionInProtein)
```

Comment:
replace this for loop with 
fragmentCoveredResidues.AddRange(psm.FragmentCovereagePositionInProtein)

Reply:
Replaced with UnionWith which is the AddRange equivalent for hash sets.